### DISCUSSION TITLE
Zustand & MongoDB with next.js

### DISCUSSION THREAD

**Discussion by PhilippWeiss85** [neuefische/questions#294] [Web] [answered]
Hello!
I have the following issue:
I have implemented zustand a couple of days ago and want to connect it now with my mongoDB. In my useStore i tried to update my appendNewGame function which adds a new gamecard to the database. It should also update my games array state in the useStore. However if I try to access the games array in my gamelist page i get the following message:
```Source
lib/dbConnect.js (6:8) @ eval

  4 | 
  5 | if (!MONGODB_URI) {
> 6 | throw new Error("Please define the MONGODB_URI environment variable inside .env.local");
    |      ^
  7 | }
  8 | 
  9 | 
```

If I try to use the SSR props (like it is implemented in the PR) my cardlist page does not re-render after submitting the form which appends a new card. It only renders the new data after a manual page reload. What can I do to resolve this issue? Should i get back to the basic state management without zustand when using mongoDB? 

Here is the link to my PR:
https://github.com/PhilippWeiss85/capstone-project/pull/33

Thanks for the help!

**Comment by ThomasF85** [comment, marked as answer]
Hi Philipp,

you cannot access your database from pages or components that are being rendered in the browser. The browser has no access to your environment variables (MONGODB_URI). You can use your dbConnect function inside your API code or inside getServerSideProps or getStaticProps.
If you want a page to update after sending a POST request and creating a new card / game there is several different ways you can do that:
1. If your API returns the newly created game object after the POST request, you can add it to your state manually. For that you need a state first (zustand or local state - both work)
2. You could use another API call to retrieve all games and put them into your state (also requires a state)
3. This seems a little hacky and I would recommend going with Option 1 instead: You can force to reload the data from the server by using next/router and `router.replace(router.asPath)`

**Reply by PhilippWeiss85** [reply]
Thanks for the reply Thomas! I've tried to solve the issue with the solution you first mentioned.
